### PR TITLE
Remove CNI configs on reset

### LINKED ISSUE #823: CNI configs left after reset
**Version**
```
$ k0s version
v0.12.0-35-gc760252
```
Checked out #779

**What happened?**
I ran `k0s reset` and ran `k0s install controller --enable-worker` and started k0s
```
k0s kc get pods -A
NAMESPACE     NAME                              READY   STATUS              RESTARTS   AGE
kube-system   coredns-5c98d7d4d8-z2rsc          0/1     ContainerCreating   0          6m19s
kube-system   konnectivity-agent-xq9xz          0/1     ContainerCreating   0          6m6s
kube-system   kube-proxy-g2s9h                  1/1     Running             0          6m16s
kube-system   kube-router-zlkf8                 1/1     Running             0          6m16s
kube-system   metrics-server-6fbcd86f7b-bhjc9   0/1     ContainerCreating   0          6m6s
```

the three pods above got stuck in `ContrainerCreating` forever. 

Remedy was to remove:
```
/etc/cni/net.d/10-calico.conflist 
/etc/cni/net.d/calico-kubeconfig 
```

It seems that reset doesn't clean up everything.

### PR DESCRIPTION
**Pull request by jasmingacic**
Fixes #823

### CODE REVIEW

File: pkg/install/cleanup_unix.go [hunk 1]
```diff
@@ -206,6 +207,33 @@ func (c *CleanUpConfig) RemoveAllDirectories() error {
...
+func removeCNILeftovers() error {
+	var msg []string
+
+	calico10Conflist := "/etc/cni/net.d/10-calico.conflist"
+	calicoKubeconfig := "/etc/cni/net.d/calico-kubeconfig"
+	kuberouterConf := "/etc/cni/net.d/10-kuberouter.conflist"
+
+	if err := os.Remove(calico10Conflist); err != nil {
+		fmtError := fmt.Errorf("failed to delete %v. err: %v", calico10Conflist, err)
+		msg = append(msg, fmtError.Error())
```
Comment (jnummelin) [inline pkg/install/cleanup_unix.go:226]: If the cluster was running kube-router CNI, this file will not exist and thus `os.Remove` will fail with:
```
remove /etc/cni/net.d/10-calico.conflist: no such file or directory
```

So trying to remove a non-existing file should not be an error in reset. Same goes for all the files of course

**Review by jnummelin — CHANGES_REQUESTED** [review]

File: pkg/install/cleanup_unix.go (hunk 1 above)
Comment (jasmingacic) [reply pkg/install/cleanup_unix.go:226]: It doesn't break the command. It just returns this info at the end
```
INFO[2021-04-08 10:17:27] deleting k0s generated data-dir (/var/lib/k0s) and run-dir (/run/k0s) 
INFO failed to delete /etc/cni/net.d/10-calico.conflist. err: remove /etc/cni/net.d/10-calico.conflist: no such file or directory, failed to delete /etc/cni/net.d/calico-kubeconfig. err: remove /etc/cni/net.d/calico-kubeconfig: no such file or directory 
INFO[2021-04-08 10:17:27] k0s cleanup operations done. To ensure a full reset, a node reboot is recommended. 
```

Comment (jasmingacic) [reply pkg/install/cleanup_unix.go:226]: which is indeed confusing

Comment (jasmingacic) [reply pkg/install/cleanup_unix.go:227]: only debug will print what was not deleted:
```
# k0s reset --debug
...
DEBU[2021-04-08 10:37:35] failed to delete /etc/cni/net.d/10-calico.conflist. err: remove /etc/cni/net.d/10-calico.conflist: no such file or directory, failed to delete /etc/cni/net.d/calico-kubeconfig. err: remove /etc/cni/net.d/calico-kubeconfig: no such file or directory, failed to delete /etc/cni/net.d/10-kuberouter.conflist. err: remove /etc/cni/net.d/10-kuberouter.conflist: no such file or directory 
```

File: pkg/install/cleanup_unix.go [hunk 2]
```diff
@@ -209,5 +210,32 @@ func (c *CleanUpConfig) RemoveAllDirectories() error {
...
+	}
+	if err := os.Remove(calicoKubeconfig); err != nil {
+		fmtError := fmt.Errorf("failed to delete %v. err: %v", calicoKubeconfig, err)
+		msg = append(msg, fmtError.Error())
```
Comment (jnummelin) [inline pkg/install/cleanup_unix.go:231]: If we only use the error for logging, why wrap it first as `error` and then use only the message from it?

Comment (jasmingacic) [reply pkg/install/cleanup_unix.go]: replaced it with `fmt.Sprintf`

**Pull request closed without merging**